ui/src/components/Bouts/BoutPage.tsx

```diff
@@ -0,0 +1,697 @@
+  }, [nearestMinutes]);
+
+  const expandTimelineEnd = useCallback(
+    (currentTime: Date) => {
+      setTimelineEndTime((timelineEndTime) =>
+        min([
+          currentTime,
+          roundToNearest(
+            addMinutes(timelineEndTime, timeBuffer),
+            nearestMinutes,
+            "ceil",
+          ),
+        ]),
+      );
+    },
+    [nearestMinutes],
+  );
+
+  const timelineStartTimeMinusADay = subDays(timelineStartTime, 1);
+  // If feed is present, and there's no pre-set time,
+  // get latest stream and last <timeBuffer> minutes of segments.
+  // Set time to end of last segment
+  const feedStreamQueryResult = useListFeedStreamsQuery(
+    {
+      feedId: feed?.id,
+      fromDateTime: timelineStartTime,
+      toDateTime: timelineEndTime,
+      dayBeforeFromDateTime: timelineStartTimeMinusADay,
+    },
+    { enabled: !!feed?.id },
+  );
+
+  // Get detections at current time plus or minus 1 hour
+  const minDetectionsTime = subMinutes(targetTime, 60);
+  const maxDetectionsTime = addMinutes(targetTime, 60);
+  const detectionQueryResult = useDetectionsQuery(
+    {
+      feedId: feed?.id,
+      filter: {
+        timestamp: {
+          greaterThanOrEqual: minDetectionsTime,
+          lessThanOrEqual: maxDetectionsTime,
+        },
+      },
+    },
+    { enabled: !!feed?.id },
+  );
+
+  const feedStreams = useMemo(
+    () => feedStreamQueryResult.data?.feedStreams?.results ?? [],
+    [feedStreamQueryResult],
+  );
+  const feedStream = feedStreams[0];
+  const feedSegments = useMemo(
+    () => feedStreams.flatMap(({ feedSegments }) => feedSegments),
+    [feedStreams],
+  );
+
+  const detections = detectionQueryResult.data?.detections?.results ?? [];
+  const [boutForm, setBoutForm] = useState<{
+    errors: Record<string, string>;
+    isSaving: boolean;
+  }>({
+    errors: {},
+    isSaving: false,
+  });
+  const createBoutMutation = useCreateBoutMutation({
+    onSuccess: ({ createBout: { errors, result } }) => {
+      if (errors && errors.length > 0) {
+        console.error(errors);
+        setBoutForm((form) => ({
+          ...form,
+          errors: {
+            ...form.errors,
+            ...Object.fromEntries(
+              errors.map(({ code, message }) => [code, message] as const),
+            ),
+          },
+        }));
+      } else if (result) {
+        router.push(`/bouts/${result.id}`);
+      }
+    },
+  });
+
+  const updateBoutMutation = useUpdateBoutMutation({
+    onSuccess: ({ updateBout: { errors } }) => {
+      if (errors && errors.length > 0) {
+        console.error(errors);
+        setBoutForm((form) => ({
+          ...form,
+          errors: {
+            ...form.errors,
+            ...Object.fromEntries(
+              errors.map(({ code, message }) => [code, message] as const),
+            ),
+          },
+        }));
+      } else {
+        setBoutSaved(true);
+        setTimeout(() => {
+          setBoutSaved(false);
+        }, 5000);
```

Comment:
Just wondering, why does this need the 5s timeout?

Reply:
Ah this one's for the 'saved bout' snack message